.gitlab-ci.yml

```diff
@@ -103,11 +104,52 @@ test:
   script:
     - cargo test --verbose --all-features --no-fail-fast --workspace
 
-codecov:
+clippy-std:
   stage:                           workspace
   <<:                              *docker-env
   needs:
     - check-std
+  script:
+    - for crate in ${ALL_CRATES}; do
+        cargo clippy --verbose --all-features --manifest-path ${crate}/Cargo.toml -- -D warnings;
+      done
+
+clippy-wasm:
+  stage:                           workspace
+  <<:                              *docker-env
+  needs:
+    - check-wasm
+  script:
+    - for crate in ${ALL_CRATES}; do
+        cargo clippy --verbose --no-default-features --manifest-path ${crate}/Cargo.toml --target wasm32-unknown-unknown -- -D warnings;
+      done
+
+fmt:
+  stage:                           workspace
+  <<:                              *docker-env
+  script:
+    - cargo fmt --verbose --all -- --check
+
+miri:
+  stage:                           extra
+  <<:                              *docker-env
+  needs:
+    - test
+  script:
+    # miri uses xargo hence can't be used with RUSTC_WRAPPER=sccache
+    # miri setup is needed because by default is demands interaction asking for dependencies
```

Comment:
This should not be needed any more, recently a made `cargo miri test` recognize the `CI` env var to avoid interactive questions.

Reply:
oh tahnks, that's good to know, is it mentioned in some docs?

Reply:
No explicitly (that doesn't seem to be common), but the docs do say how we [recommend to set up CI](https://github.com/rust-lang/miri/#running-miri-on-ci).